### PR TITLE
Add first pass at code owners

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

Adds CODEOWNERS for more explicit ownership

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1 +1,40 @@
...
+
+typescript/sdk @yorhodes @jmrossy
+typescript/sdk/src/gas @tkporter
+typescript/sdk/src/middleware @yorhodes
+typescript/sdk/src/providers @jmrossy
+typescript/sdk/src/deploy @yorhodes
+
+## Apps
+typescript/token @yorhodes @jmrossy
+typescript/helloworld @nambrot
```
Comment (nambrot) [inline .github/CODEOWNERS:36]: Happy to do this, but curious how I got this one :)

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1 +1,40 @@
...
+
+solidity/ @yorhodes @asaj @tkporter
+solidity/contracts/igps @tkporter
+solidity/contracts/isms @asaj
+solidity/contracts/middleware @yorhodes
+
+## Agents
+
+rust/ @mattiecnvr @tkporter @asaj
+rust/agents/relayer @mattiecnvr
```
Comment (nambrot) [inline .github/CODEOWNERS:20]: Not sure we need to be this granular?

**Review by nambrot — APPROVED** [review]

File: .github/CODEOWNERS (hunk 2 above)
Comment (asaj) [reply .github/CODEOWNERS:20]: +1, slight preference for "package" level owners, e.g.:

`rust`, `sdk`, `solidity`

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -1 +1,40 @@
+# File extension owners
+
+*.sol @yorhodes @asaj @tkporter
```
Comment (asaj) [inline .github/CODEOWNERS:3]: How do these interact with the owners below? I presume union?

**Review by asaj — APPROVED** [review]
Thanks for doing this. Are you doing the other repos as well or should I?

File: .github/CODEOWNERS (hunk 3 above)
Comment (yorhodes) [reply .github/CODEOWNERS:3]: later overrides earlier

**Comment by yorhodes** [comment]
> Thanks for doing this. Are you doing the other repos as well or should I?

https://github.com/hyperlane-xyz/hyperlane-token/commit/b009989328333a3d725979491b2897db2b9473b9
https://github.com/hyperlane-xyz/docs/commit/f07f96b8025b6ba9cc7114f6257143c2c3fb87c0
https://github.com/hyperlane-xyz/hyperlane-app-template/commit/5aac31651bd0340839315011457f5a12388e7d48
https://github.com/hyperlane-xyz/hyperlane-explorer/commit/1da9de17aec086993e6e58d6c6ecf0c35f0f44d2
https://github.com/hyperlane-xyz/hyperlane-widgets/commit/dc793abcac2618f571d841cec682b076061086e5
https://github.com/hyperlane-xyz/hyperlane-warp-ui-template/commit/b4f0fcbd60181f98ff50239c0a2a7903018b4a5a

Need reviews here
https://github.com/hyperlane-xyz/hyperlane-deploy/pull/32
https://github.com/hyperlane-xyz/hyperlane-quickstart/pull/3

**Pull request merged**